superbench/analyzer/data_diagnosis.py

```diff
@@ -93,25 +94,53 @@ def _get_baseline_of_metric(self, baseline, metric):
                 logger.warning('DataDiagnosis: get baseline - {} baseline not found'.format(metric))
                 return -1
 
-    def _get_criteria(self, rule_file, baseline_file):
-        """Get and generate criteria of metrics.
+    def __get_metrics_in_criteria(self, rule, benchmark_rules, baseline):
+        """Get metrics with baseline in the rule.
 
-        Read rule file and baseline file. For each rule, use metric with regex
-        in the metrics of the rule to match the metric full name from raw data
-        for each benchmark in the rule, and then merge baseline and rule for
-        matched metrics.
+        Use metric with regex in the metrics of the rule to match
+        the metric full name from raw data for each benchmark in the rule,
+        and then merge baseline for each metric.
 
         Args:
-            rule_file (str): The path of rule yaml file
-            baseline_file (str): The path of baseline json file
+            rule (str): the name of the rule
+            benchmark_rules (dict): the dict of rules
+            baseline (dict): the dict of baseline of metrics
+        """
+        if self._sb_rules[rule]['function'] == 'multi_rules':
+            return
+        single_rule_metrics = benchmark_rules[rule]['metrics']
+        benchmark_metrics = self._get_metrics_by_benchmarks(single_rule_metrics)
+        for benchmark_name in benchmark_metrics:
+            if benchmark_name not in self._metrics:
+                logger.warning('DataDiagnosis: get criteria failed - {}'.format(benchmark_name))
+                continue
+            # get rules and criteria for each metric
+            for metric in self._metrics[benchmark_name]:
+                # metric full name in baseline
+                if metric in single_rule_metrics:
+                    self._sb_rules[rule]['metrics'][metric] = self._get_baseline_of_metric(baseline, metric)
+                    self._enable_metrics.append(metric)
+                    continue
+                # metric full name not in baseline, use regex to match
+                for metric_regex in benchmark_metrics[benchmark_name]:
+                    if re.search(metric_regex, metric):
+                        self._sb_rules[rule]['metrics'][metric] = self._get_baseline_of_metric(baseline, metric)
+                        self._enable_metrics.append(metric)
```

Comment:
How to handle duplicated metric?

Reply:
why there are duplicated metrics, each metric should be unique I think

Reply:
Actually, I don't know the major logic here. I think `_enable_metrics` should store the metric itself. If I write two benchmark_name have overlapping, for example, `xx/xx/*` and `xx/xx/xxx*`, I think it will have the bug in your code.